float32 corner_x
float32 corner_y
float32 corner_z
